src/code/fault.c

```diff
@@ -683,87 +782,115 @@ void Fault_WaitForButtonCombo() {
     }
 }
 
-void Fault_DrawMemDumpPage(const char* title, u32* addr, u32 param_3) {
-    u32* alignedAddr;
+void Fault_DrawMemDumpContents(const char* title, u32 addr, u32 arg2) {
+    u32 alignedAddr = addr;
     u32* writeAddr;
     s32 y;
     s32 x;
 
-    alignedAddr = addr;
-
-    if (alignedAddr < (u32*)0x80000000) {
-        alignedAddr = (u32*)0x80000000;
+    // Ensure address is within the bounds of RDRAM (Fault_DrawMemDump has already done this)
+    if (alignedAddr < K0BASE) {
+        alignedAddr = K0BASE;
     }
-    if (alignedAddr > (u32*)0x807FFF00) {
-        alignedAddr = (u32*)0x807FFF00;
+    if (alignedAddr > K0BASE + 0x800000 - 0x100) {
+        alignedAddr = K0BASE + 0x800000 - 0x100;
```

Comment:
`0x800000` is 8mb RAM, `0x100` is just so you can't position the page start at the very end of RAM.

Reply:
Ah right. That sounds like it's worth a comment, then.

Reply:
if I'm reading the for loops right, the page draws `4*22` words, that is `4*22*4 = 352` bytes
so if it starts `0x100 = 256` before the end of ram it will read outside of ram, do you know what happens on hardware then? (I assume it reads 0 or garbage from unmapped addresses?)

Reply:
It's either 0 or an open bus read, which is `((address & 0xFFFF) << 0x10 ) | (address & 0xFFFF)`

Reply:
Thanks to @fig02 for checking this on hardware, the result is it reads 0.